src/garth/data/body_battery.py

```diff
@@ -0,0 +1,304 @@
+            return None
+        return readings[-1].level - readings[0].level
+
+    @classmethod
+    def get(
+        cls,
+        date_str: str | date | None = None,
+        *,
+        client: http.Client | None = None,
+    ) -> Self | None:
+        """Get complete Body Battery and stress data for a specific date."""
+        client = client or http.client
+        date_str = format_end_date(date_str)
+
+        path = f"/wellness-service/wellness/dailyStress/{date_str}"
+        response = client.connectapi(path)
+
+        if not isinstance(response, dict):
+            return None
+
+        # Parse timestamps
+        start_gmt = datetime.fromisoformat(
+            response.get("startTimestampGMT", "").replace("Z", "+00:00")
+        )
+        end_gmt = datetime.fromisoformat(
+            response.get("endTimestampGMT", "").replace("Z", "+00:00")
+        )
+        start_local = datetime.fromisoformat(
+            response.get("startTimestampLocal", "").replace("Z", "+00:00")
+        )
+        end_local = datetime.fromisoformat(
+            response.get("endTimestampLocal", "").replace("Z", "+00:00")
+        )
+
+        return cls(
+            user_profile_pk=response.get("userProfilePK", 0),
+            calendar_date=response.get("calendarDate", ""),
+            start_timestamp_gmt=start_gmt,
+            end_timestamp_gmt=end_gmt,
+            start_timestamp_local=start_local,
+            end_timestamp_local=end_local,
+            max_stress_level=response.get("maxStressLevel", 0),
+            avg_stress_level=response.get("avgStressLevel", 0),
+            stress_chart_value_offset=response.get(
+                "stressChartValueOffset", 0
+            ),
+            stress_chart_y_axis_origin=response.get(
+                "stressChartYAxisOrigin", 0
+            ),
+            stress_values_array=response.get(
+                "stressValuesArray", []
+            ),
+            body_battery_values_array=response.get(
+                "bodyBatteryValuesArray", []
+            )
+        )
+
+    @classmethod
+    def list(
+        cls,
+        end: date | str | None = None,
+        period: int = 1,
+        *,
+        client: http.Client | None = None,
+    ) -> List[Self]:
+        """Get Body Battery and stress data for multiple days."""
+        end_date = format_end_date(end)
+        all_data = []
+
+        for i in range(period):
+            date_to_fetch = end_date - timedelta(days=i)
+            data = cls.get(date_to_fetch, client=client)
+            if data:
+                all_data.append(data)
+
+        return all_data
+
+
+@dataclass
+class BodyBatteryData:
+    """Legacy Body Battery events data (sleep events only)."""
+    event: BodyBatteryEvent | None = None
+    activity_name: str | None = None
+    activity_type: str | None = None
+    activity_id: str | None = None
+    average_stress: float | None = None
+    stress_values_array: List[List[int]] | None = None
```

Comment:
please use `list[]` instead of `List[]` for consistency

Reply:
done